docs_nnx/nnx_glossary.rst

```diff
@@ -0,0 +1,43 @@
+
+    PRNG states
+      A Flax :class:`nnx.Module <flax.nnx.Module>` can keep a reference of a `pseudorandom number generator (PRNG) <https://jax.readthedocs.io/en/latest/random-numbers.html>`__ state object :class:`nnx.Rngs <flax.nnx.Rngs>` that can generate new JAX `PRNG <https://en.wikipedia.org/wiki/Pseudorandom_number_generator>`__ keys. These keys are used to generate random JAX arrays through `JAX's functional pseudorandom number generators <https://jax.readthedocs.io/en/latest/random-numbers.html>`__.
+      You can use a PRNG state with different seeds to add more fine-grained control to your model (for example, to have independent random numbers for parameters and dropout masks).
+      Refer to the `Flax Randomness/PRNG guide <https://flax.readthedocs.io/en/latest/guides/randomness.html>`__
+      for more details.
+
+    Split and merge
+      :meth:`nnx.split <flax.nnx.split>` is a way to represent an :class:`nnx.Module <flax.nnx.Module>` by two parts: 1) a static Flax NNX :term:`GraphDef <GraphDef>` that captures its Pythonic static information; and 2) one or more :term:`Variable state(s)<Variable state>` that capture its `JAX arrays <https://jax.readthedocs.io/en/latest/_autosummary/jax.Array.html>`__ (``jax.Array``) in the form of `JAX pytrees <https://jax.readthedocs.io/en/latest/working-with-pytrees.html>`__. They can be merged back to the original ``nnx.Module`` using :meth:`nnx.merge <flax.nnx.merge>`.
+
+    Transformation
+      A Flax NNX transformation (transform) is a Flax NNX-wrapped version of a `JAX transformation <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ that allows a transformed function to take the Flax NNX :term:`Module<Module>` (``nnx.Module``) as input or output. For example, a "lifted" version of `jax.jit <https://jax.readthedocs.io/en/latest/_autosummary/jax.jit.html#jax.jit>`__ is :meth:`nnx.jit <flax.nnx.jit>`. Check out the `Flax NNX transforms guide <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ to learn more.
+
+    Variable
+      The weights / parameters / data / array :class:`Variable <flax.nnx.Variable>` residing in a Flax :term:`Module<Module>`. Variables are defined inside modules as :class:`nnx.Variable <flax.nnx.Variable>` or its subclasses.
+
+    Variable state
+      :class:`nnx.VariableState <flax.nnx.VariableState>` is a purely functional `JAX pytree <https://jax.readthedocs.io/en/latest/working-with-pytrees.html>`__ of all the :term:`Variables<Variable>` inside a :term:`Module<Module>`. Since it is pure, it can be an input or output of a `JAX transformation <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ function. ``VariableState`` is obtained by using :meth:`nnx.split <flax.nnx.split>` and :meth:`nnx.merge <flax.nnx.merge>` on the :class:`nnx.Module <flax.nnx.Module>`. (Refer to  :term:`splitting<Split and merge>` and :term:`Module<Module>`.)
```

Comment:
> ``VariableState`` is obtained by using :meth:`nnx.split <flax.nnx.split>` and :meth:`nnx.merge <flax.nnx.merge>`

It's only obtained by using `nnx.split`. Remove the mention of `merge`.

Reply:
SGTM!

"...by using `nnx.split` on the `nnx.Module`..."

```
.... Since it is pure, it can be an input or output of a `JAX transformation <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ function. ``VariableState`` is obtained by using :meth:`nnx.split <flax.nnx.split>` on the :class:`nnx.Module <flax.nnx.Module>`. (Refer to  :term:`splitting<Split and merge>` and :term:`Module<Module>`.)
```